src/lib.rs

```diff
@@ -96,10 +96,11 @@ impl<Data: AsRef<[u8]>> Reader<Data> {
     }
 
     /// Iterator over the texture's mip levels
-    pub fn levels(&self) -> impl ExactSizeIterator<Item = &[u8]> + '_ {
-        self.level_index()
-            .unwrap()
-            .map(move |level| &self.input.as_ref()[level.offset as usize..(level.offset + level.length_bytes) as usize])
+    pub fn levels(&self) -> impl ExactSizeIterator<Item = Level> + '_ {
+        self.level_index().unwrap().map(move |level| Level {
+            bytes: &self.input.as_ref()[level.offset as usize..(level.offset + level.length_bytes) as usize],
```

Comment:
Why the as_ref() call here?

Reply:
You mean the pre-existing one?

Reply:
This is because that's the only trait we have implemented for `Data`: https://github.com/BVE-Reborn/ktx2/blob/808b212deaf518ce86cfa53c283e0429877cabcf/src/lib.rs#L40